go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -41,11 +41,7 @@ func (npw *nodePortWatcher) AddService(service *kapi.Service) {
 	for _, svcPort := range service.Spec.Ports {
 		protocol := strings.ToLower(string(svcPort.Protocol))
 		protocol6 := protocol + "6"
-		if util.ServiceTypeHasNodePort(service) {
-			if err := util.ValidatePort(svcPort.Protocol, svcPort.NodePort); err != nil {
-				klog.Errorf("Skipping service add for svc: %s, err: %v", svcPort.Name, err)
-				continue
-			}
+		if svcPort.NodePort > 0 {
```

Comment:
Why not keep the same check as before: `util.ServiceTypeHasNodePort(service) `?

Reply:
because that is making a wrong assumption, a LoadBalancer service doesn' t need nodePort, it is a legacy thing and in recent versions that can be surpressed, the right check is to see if the NodePort is set